test/functional.jl

```diff
@@ -374,6 +374,20 @@ import Base.flatten
 @test eltype(flatten(UnitRange{Int8}[1:2, 3:4])) == Int8
 @test_throws ArgumentError collect(flatten(Any[]))
 
+
+# concat
+# ------
+
+import Base.concat
+
+@test collect((concat((i:i+10 for i in 1:3)))) == [1:11 2:12 3:13]
```

Comment:
You could write it like this `collect(concat(i:i+10 for i in 1:3))` wouldn't you?